### PR TITLE
feat:updated task

### PR DESCRIPTION
**Pull request by juandgaines**
- updated task to done
- feedback messages on agenda

### CODE REVIEW

File: agenda/data/src/main/java/com/juandgaines/agenda/data/task/DefaultTaskRepository.kt [hunk 1]
```diff
@@ -73,26 +74,27 @@ class DefaultTaskRepository @Inject constructor(
...
+                taskDao.upsertTask(it.toTaskEntity())
+            }.await()
+        }
 
+        return taskDao.getTaskById(taskId)?.toTask()?.let {
+            Result.Success(it)
+        } ?: run {
+            Result.Error(LocalError.NOT_FOUND)
+        }
     }
```
Comment (Kai-Cilliers) [inline agenda/data/src/main/java/com/juandgaines/agenda/data/task/DefaultTaskRepository.kt:95]: Here, you will be suspended until you get a response from the API. Some apps might require this behavior, so I am not saying this is incorrect. Still, another approach would be just to return the value in the DAO, considering that if you have the ID, you’ve synced your DB with the remote agenda items, and it's safe to assume the data would be there.

You could argue that you may not have the latest info if the item was modified on a different device, which is a fair consideration. In that case, I’d return a `Flow.` It will immediately return the local result, and then you can asynchronously fetch the data from the API and update your DB, which will trigger a new emission.

In short, considering the app is offline-first, I’d go with keeping it simple and simply returning the data from the DB. This keeps you from over-optimising details or features that don’t bring much value to your clients.

File: agenda/data/src/main/java/com/juandgaines/agenda/data/agenda/DefaultAgendaRepository.kt [hunk 2]
```diff
@@ -55,4 +61,12 @@ class DefaultAgendaRepository @Inject constructor(
...
         }
     }.asEmptyDataResult()
+
+    override suspend fun updateTask(taskId:String): EmptyDataResult<DataError> {
+        val response= taskRepository.getTaskById(taskId)
+            .onSuccess {
+                taskRepository.updateTask(it.copy(isDone = !it.isDone))
+            }
+        return response.asEmptyDataResult()
+    }
```
Comment (Kai-Cilliers) [inline agenda/data/src/main/java/com/juandgaines/agenda/data/agenda/DefaultAgendaRepository.kt:71]: Considering this is an offline-first app, would it not make more sense to pass in the full model to update the DB? With your current implementation, your app won’t be able to save a task when it does not have an internet connection

File: agenda/data/src/main/java/com/juandgaines/agenda/data/task/DefaultTaskRepository.kt (hunk 1 above)
Comment (juandgaines) [reply agenda/data/src/main/java/com/juandgaines/agenda/data/task/DefaultTaskRepository.kt:95]: got it

File: agenda/data/src/main/java/com/juandgaines/agenda/data/agenda/DefaultAgendaRepository.kt (hunk 2 above)
Comment (juandgaines) [reply agenda/data/src/main/java/com/juandgaines/agenda/data/agenda/DefaultAgendaRepository.kt:71]: good catch, i will change it
